### ISSUE TITLE
Add BenchmarkCI

### ISSUE THREAD

**Issue by mateuszbaran** [JuliaNLSolvers/Manifolds.jl#118]
It would be nice to easily detect possible performance regressions.

Link: https://github.com/tkf/BenchmarkCI.jl

**Comment by kellertuer** [comment]
Since this just came up in #545 : Should we add these here (we could adapt a few from the preprint?) Or should we do a separate repository that could also cover the python/matlab tests?

**Comment by mateuszbaran** [comment]
I'm not sure, in any case I wouldn't mix CI benchmarks and benchmarks that compare different libraries.

**Comment by mateuszbaran** [comment]
Yes, I'd like to add those benchmarks. I think we didn't see much problems so far because I was paying attention to performance during reworks and reviews but I'd prefer to have a benchmark suite as well.
